core/src/main/java/de/danoeh/antennapod/core/service/download/DownloadService.java

```diff
@@ -190,10 +193,26 @@ public void onCreate() {
         cancelDownloadReceiverFilter.addAction(ACTION_CANCEL_ALL_DOWNLOADS);
         cancelDownloadReceiverFilter.addAction(ACTION_CANCEL_DOWNLOAD);
         registerReceiver(cancelDownloadReceiver, cancelDownloadReceiverFilter);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            this.registerLollipopNetworkListener();
+        }
 
         downloadCompletionThread.start();
     }
 
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    void registerLollipopNetworkListener() {
+        ConnectionStateMonitor connectionMonitor = new ConnectionStateMonitor();
```

Comment:
Even though `enable` and `disable` are stateless I would prefer to avoid having to create new instances on each call and have the `connectionMonitor` instance stored in the service instead. From outside it is not clear why we should not use the same instance on every call.